### PR TITLE
Constraint hub with JSON store backing

### PR DESCRIPTION
**Pull request by richard-rogers**
## Description

Updates constraints hub hack-a-thon work. Adds a `ConfigStore` that stores constraints as model metadata using the JSON store (unversioned). 

- [ ] I have reviewed the [Guidelines for Contributing](CONTRIBUTING.md) and the [Code of Conduct](CODE_OF_CONDUCT.md).

### CODE REVIEW

File: python/whylogs/experimental/api/constraints/__init__.py [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+    column_is_probably_unique,
+    condition_meets,
+#    ConditionMeetsParams,
+    count_below_number,
+#    CountBelowNumber,
+    distinct_number_in_range,
+#    DistinctNumberInRangeParams,
+    is_in_range,
+    is_non_negative,
+#    IsNonNegativeParams,
```
Comment (murilommen) [inline python/whylogs/experimental/api/constraints/__init__.py]: why keep these commented?

File: python/whylogs/experimental/extras/configstore.py [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+            output = os.popen(f"cd {self._path}; {command}")
+            logger.debug(f"{command}: {output.read()}")
+
+    def __init__(self, org_id: str, dataset_id: str, what: str, create: bool = False, repo_path: str = "/tmp"):
+        self._repo_path = repo_path
+        self._path = f"{self._repo_path}/{org_id}/{dataset_id}"
+        super().__init__(org_id, dataset_id, what, create)
+
+    def create(self) -> None:
+        if not os.path.isdir(self._path):
```
Comment (murilommen) [inline python/whylogs/experimental/extras/configstore.py:87]: is this always a path to a directory?

File: python/whylogs/experimental/api/constraints/__init__.py [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+}
+"""
+
+def assemble_constraint(constraint_dict: dict) -> Optional[TypedDict]:
+    constraint_name = None
+    if constraint_dict.get("expression") and constraint_dict.get("name"):
+        condition = PrefixCondition(constraint_dict.get("expression"))
+        returned_constraint = DatasetConstraint(condition=condition, name=constraint_dict.get("name"))
+        return returned_constraint
+    """
```
Comment (murilommen) [inline python/whylogs/experimental/api/constraints/__init__.py]: why is this code block sitting here?

File: python/whylogs/experimental/api/constraints/__init__.py (hunk 1 above)
Comment (richard-rogers) [reply python/whylogs/experimental/api/constraints/__init__.py]: I haven't moved them over from the hackathon branch yet.

File: python/whylogs/experimental/api/constraints/__init__.py (hunk 3 above)
Comment (richard-rogers) [reply python/whylogs/experimental/api/constraints/__init__.py]: Also not yet fully moved over from the hackathon branch

File: python/whylogs/experimental/extras/configstore.py (hunk 2 above)
Comment (richard-rogers) [reply python/whylogs/experimental/extras/configstore.py:87]: Yes. A `RemoteGitConfigStore` wouldn't be much harder.

File: python/tests/experimental/api/constraints/test_constraints_yaml.py [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+    generate_constraints_from_reference_profile,
+)
+
+yaml_string = """\
+
+id: "model-23"
+version: 1
+hash: abcabc231 # maybe from git?
+
+id: "model-23"
```
Comment (FelipeAdachi) [inline python/tests/experimental/api/constraints/test_constraints_yaml.py:21]: the following 3 lines are duplicated?

File: python/tests/experimental/extras/test_configstore.py [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+        assert cs.exists()
+        assert os.path.isdir(f"{prefix}/my_org/my_model")
+        assert os.path.isfile(f"{prefix}/my_org/my_model/test")
+        assert cs.get_available_versions() == [Version("0.0.0")]
+        assert cs.get_version_of_latest() == Version("0.0.0")
+        assert cs.get_latest() == ""
+
+        # repeat with existing directory
+        cs = LocalGitConfigStore("my_org", "my_model", "test", repo_path=prefix)
+        assert cs.exists()
```
Comment (FelipeAdachi) [inline python/tests/experimental/extras/test_configstore.py:24]: what happens if we call cs.create() on an existing config store? should we create a test for it?

File: python/whylogs/experimental/extras/configstore.py [hunk 6]
```diff
@@ -0,0 +1,203 @@
...
+            f'git commit -m "promoting {new_version}"',
+            "git checkout master",
+            f"git merge {new_version}",
+            f"git branch -d {new_version}",
+            f"git tag {new_version}",
+        ]
+        self._run_commands(cmds)
+
+
+class JsonStore(ConfigStore):
```
Comment (FelipeAdachi) [inline python/whylogs/experimental/extras/configstore.py:152]: if this is Whylabs-specific, shouldn't we include WhyLabs somewhere in the name?

**Pull request closed without merging**